File: app/src/main/java/com/druger/aboutwork/interfaces/view/CompanyDetailView.java
package com.druger.aboutwork.interfaces.view;

import com.arellomobile.mvp.MvpView;
import com.druger.aboutwork.model.CompanyDetail;
import com.druger.aboutwork.model.Review;

import java.util.List;

/**
 * Created by druger on 01.05.2017.
 */

public interface CompanyDetailView extends MvpView, NetworkView {

    void updateAdapter();

    void showReviews(List<Review> reviews);

    void showCompanyDetail(CompanyDetail company);

    void showProgressReview();

    void hideProgressReview();
}

File: app/src/main/java/com/druger/aboutwork/interfaces/view/CompanyDetailView.kt
package com.druger.aboutwork.interfaces.view

import androidx.annotation.DrawableRes
import com.druger.aboutwork.model.CompanyDetail
import com.druger.aboutwork.model.Review
import moxy.MvpView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

/**
 * Created by druger on 01.05.2017.
 */

@StateStrategyType(OneExecutionStateStrategy::class)
interface CompanyDetailView : MvpView, NetworkView {

    fun updateAdapter()

    fun showReviews(reviews: List<Review>, isFilter: Boolean = false)

    fun showCompanyDetail(company: CompanyDetail)

    fun showAuth()

    fun addReview()

    fun showFilterDialog(position: String, city: String)

    fun setFilterIcon(@DrawableRes icFilter: Int)

    fun showEmptyReviews(isFilter: Boolean = false)
}
